snapcraft/internal/elf.py

```diff
@@ -513,6 +534,10 @@ def get_elf_files(root: str,
     """
     elf_files = set()  # type: Set[ElfFile]
 
+    # We need to reset the empty soname cache entries in case they are provided
+    # by a new set of root files
+    reset_soname_paths(only_none=True)
```

Comment:
How about calling the argument `empty_entries_only`? That's the wording used in the comment and much more descriptive to my mind than referring to how it's implemented.